### PR TITLE
Multisig account customization

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: aptos-move/framework/supra-framework/sources/multisig_account.move [hunk 1]
```diff
@@ -768,6 +800,15 @@ module supra_framework::multisig_account {
         };
     }
 
+    /// Update the timeout duration for the multisig account.
+    entry fun update_timeout_duration(
+        multisig_account: &signer, timeout_duration: u64) acquires MultisigAccount {
```
Comment (ghost) [inline aptos-move/framework/supra-framework/sources/multisig_account.move:814]: TODO: check the minimal.

File: aptos-move/framework/supra-framework/sources/multisig_account.move [hunk 2]
```diff
@@ -253,6 +264,12 @@ module supra_framework::multisig_account {
         new_metadata: SimpleMap<String, vector<u8>>,
     }
 
+    /// Event emitted when a transaction's timeout duration is updated.
+    struct TimeoutDurationUpdatedEvent has drop, store {
+        old_timeout_duration: u64,
+        new_timeout_duration: u64,
```
Comment (sjoshisupra) [inline aptos-move/framework/supra-framework/sources/multisig_account.move]: Also add the `address` of the multisig account as part of the event.

File: aptos-move/framework/supra-framework/sources/multisig_account.move [hunk 3]
```diff
@@ -344,7 +368,8 @@ module supra_framework::multisig_account {
         let transaction = table::borrow(&multisig_account_resource.transactions, sequence_number);
         let (_, num_rejections) = num_approvals_and_rejections(&multisig_account_resource.owners, transaction);
         sequence_number == multisig_account_resource.last_executed_sequence_number + 1 &&
-            num_rejections >= multisig_account_resource.num_signatures_required
+            (num_rejections >= multisig_account_resource.num_signatures_required ||
+            multisig_account_resource.timeout_duration < now_seconds() - transaction.creation_time_secs)
```
Comment (sjoshisupra) [inline aptos-move/framework/supra-framework/sources/multisig_account.move]:
```suggestion
           ( (num_rejections >= multisig_account_resource.num_signatures_required) ||
            (multisig_account_resource.timeout_duration < now_seconds() - transaction.creation_time_secs))
```

File: aptos-move/framework/supra-framework/sources/multisig_account.move [hunk 4]
```diff
@@ -768,6 +808,22 @@ module supra_framework::multisig_account {
         };
     }
 
+    /// Update the timeout duration for the multisig account.
+    entry fun update_timeout_duration(
+        multisig_account: &signer, timeout_duration: u64) acquires MultisigAccount {
+        assert_multisig_account_exists(address_of(multisig_account));
+        let multisig_account_resource = borrow_global_mut<MultisigAccount>(address_of(multisig_account));
+        let old_timeout_duration = multisig_account_resource.timeout_duration;
+        multisig_account_resource.timeout_duration = timeout_duration;
```
Comment (sjoshisupra) [inline aptos-move/framework/supra-framework/sources/multisig_account.move:819]: assert that `timeout_duration >= MINIMAL_TIMEOUT_DURATION` before updating it.

**Review by sjoshisupra — CHANGES_REQUESTED** [review]

File: aptos-move/framework/supra-framework/sources/multisig_account.move (hunk 2 above)
Comment (ghost) [reply aptos-move/framework/supra-framework/sources/multisig_account.move]: Done.

File: aptos-move/framework/supra-framework/sources/multisig_account.move (hunk 4 above)
Comment (ghost) [reply aptos-move/framework/supra-framework/sources/multisig_account.move:819]: Done.

**Review by sjoshisupra — APPROVED** [review]

**Pull request merged**